pkg/daemon/ceph/client/pool.go

```diff
@@ -418,6 +421,57 @@ func CreateReplicatedPoolForApp(context *clusterd.Context, clusterInfo *ClusterI
 	}
 
 	logger.Infof("creating replicated pool %s succeeded", poolName)
+
+	if checkFailureDomain {
+		if err = ensureFailureDomain(context, clusterInfo, clusterSpec, poolName, pool); err != nil {
+			return nil
+		}
+	}
+	return nil
+}
+
+func ensureFailureDomain(context *clusterd.Context, clusterInfo *ClusterInfo, clusterSpec *cephv1.ClusterSpec, poolName string, pool cephv1.PoolSpec) error {
+	if pool.FailureDomain == "" {
+		logger.Debugf("Skipping check for failure domain on pool %q as it is not specified", poolName)
+		return nil
+	}
+
+	logger.Debugf("Checking that pool %q has the failure domain %q", poolName, pool.FailureDomain)
+	details, err := GetPoolDetails(context, clusterInfo, poolName)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get pool %q details", poolName)
+	}
+
+	if details.FailureDomain == pool.FailureDomain {
```

Comment:
What if the pool has an existing rule and a new one is applied? Should we remove the old rule? Or at least print a message that the previous rule might need to be manually removed?

Reply:
I thought about removing the old rule, but decided not to in case some other pool was using the same rule. Are there any side effects of old rules staying in the crush map? I'll at least log it though.